### ISSUE TITLE
Allow same translation strings in some languages for Reused translation

### ISSUE THREAD

**Issue by maboroshin** [WeblateOrg/weblate#9450]
### Describe the issue

Kanji characters are not case-sensitive, so the following translation results are same.

* Unsubscribe -> 登録解除
* unsubscribe -> 登録解除

In addition, the Japanese language doesn't often use the plural form except for human. So following will be the same.

* Playlist -> 再生リスト
* Playlists -> 再生リスト 

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

See "Checks"

### Expected behavior

First, Please stop the "Reused translation" test in Japanese. At present, most are false positives. The same will happen with Japanese, Chinese, Korean, Thai.

To enable "Reused translation" for Japanese, at least the following implementations are required. An algorithm is needed to pass the detection of initial capital letter and plural form.

You can also limit the scope. e.g. Original English text without spaces. Perhaps faster execution of the algorithm. And it will eliminate most false positives.

**Case-sensitive detection**

* TranslationString1 and TranslationString2 are the same AND
* * The first letter of EnglishString1 is capitalized && EnglishString2 is all lowercase OR
* *  The first letter of EnglishString1 is capitalized && EnglishString2 is all uppercase

If the translation strings are same in the above cases, the test passes.

**Plural detection**

* TranslationString1 and TranslationString2 are the same AND
* * EnglishString2 is just "s" (or "es") added to EnglishString1.

If the translation strings are same in the above cases, the test passes.

If possible, the following will also be implemented
* * If EnglishString1 ends with "y", delete this "y" and add "ies". This equals EnglishString2. OR
* * If EnglishString1 ends with "f" (or "fe"), delete this "f" and add "ves". This equals EnglishString2.

### How do you run Weblate?

weblate.org service

## Scope of Impact

It would also be possible to switch detection by language after translation.

**Plural**

Few languages have fewer plurals. [A Japanese paper](https://core.ac.uk/download/pdf/292915302.pdf)* states: Many in Southeast Asia, including Japanese, Korean, Chinese, Thai, Vietnamese, and Indonesian. Perhaps this doesn't enumerate all.

Based on the above I searched again. Chinese is like Japanese may be used plurals for humans.  Korean uses plurals more frequently for humans. The Indonesian plural repeats the singular twice.

* Paper's title : "東アジア諸語における＜数＞に関する発想と表現"

**Case-sensitive**
There are numerous case-insensitive languages that can be found [here](https://ja.wiktionary.org/wiki/%E4%BB%98%E9%8C%B2:%E5%90%84%E7%A8%AE%E8%A8%80%E8%AA%9E%E3%81%AE%E5%A4%A7%E6%96%87%E5%AD%97%E3%81%AE%E4%BD%BF%E7%94%A8). In other words, false positives for case-sensitive are quite frequent.

Japanese, Chinese, Korean, Thai, Sanskrit, Hindi, Hebrew, Persian.

Also, Assamese, Arabic, Uighur, Urdu, Kannada, Gujarati, Khmer, Georgian, Tibetan, Telugu, Hausa, Pashto, Punjabi, Bengali, Marathi, Malayalam, Rao. Perhaps this doesn't enumerate all. It may not be accurate.

Based on the above I searched again. I have not re-searched all of them. Tibetan has capital letters.

**Use Unicode set**

Detection by Unicode is complex. The following conditions are required.

* TranslationString1 and TranslationString2 are the same AND
* * The first letter of EnglishString1 is capitalized && EnglishString2 is all lowercase AND
* * * The translation string contains characters in the following numeric range in Unicode

Case-insensitive character set.  Perhaps this doesn't enumerate all. [Unicode Blocks](https://unicode.org/Public/UNIDATA/Blocks.txt)
* Cyrillic [ 1024 - 1279 ]
* Armenian [ 1280 - 1423 ]
* Greek [ 880 - 1023 ]
* Hebrew [ 1424 - 1535 ]
* Arabic [ 1536 - 1791 ]
* Devanagari [ 1792 - 2431 ]
* Thai [ 3456 - 3711 ]
* CJK Symbols and Punctuation [ 10176 - 12351 ]
* Hiragana [ 12352 - 12447 ]
* Katakana [ 12448 - 12543 ]
* Enclosed CJK Letters and Months [ 12704 - 13055 ]
* CJK Unified Ideographs (Kanji) [ 13312 - 40959 ]
* Hangul Syllables [ 40960 - 55203 ]

**Comment by nijel** [comment]
The more complex logic is not really doable right now for performance reasons. The check is offloaded to the database, and doing such complex transformations would break it. Also, Weblate supports any source language, not only English…

**Comment by maboroshin** [comment]
Thanks for the reply.

I often translate from English. A simple algorithm would reduce false positives by 70-80%.

**Comment by yarons** [comment]
Hi, the fix has nothing to do with the request, now the check is being ignored for languages with a single plural form.

If you can create a list of ISO631-2/3 codes for manual exclusion we will add the codes manually instead of checking the limitations of the languages on the fly.

Thanks :)

**Comment by nijel** [comment]
Correct, I somehow missed that part. We really need a way to do this programmatically...

**Comment by yarons** [comment]
I just figured out there are some unique cases which I'm not sure we can handle programmatically but "Synchronization vs. Sync." is a good example for a case where in several languages (such as Hebrew) there is no abbreviation so these 2 strings must have the same translation regardless of screen space (sadly).

**Comment by maboroshin** [comment]
The way you describe it would require building word lists. It would be a huge project. It's not suitable for simple implementation.

A simple implementation goes like this.
- Create a list of countries to exclude
- Easily convert and compare current words
